SDK/AppCenter/Microsoft.AppCenter.Windows.Shared/Ingestion/Http/RetryableIngestion.cs

```diff
@@ -1,80 +1,143 @@
 ﻿using System;
 using System.Collections.Generic;
+using System.Linq;
 using System.Threading;
-using System.Threading.Tasks;
 using Microsoft.AppCenter.Ingestion.Models;
 
 namespace Microsoft.AppCenter.Ingestion.Http
 {
-    public class RetryableIngestion : IngestionDecorator
+    internal sealed class RetryableIngestion : IngestionDecorator
```

Comment:
Use `Timer` to avoid long-running tasks;
All logic contains in `Ingestion` class now instead of mess with `ServiceCall`s and `Ingestion`s decorators;
Many changes caused by the necessary change of Call method signature;

Reply:
Is that what the fix is about (disregarding all the refactoring) to use Timers instead of tasks?

Reply:
Here - yes. In other places, it is event + callback instead of `System.Task`